docs/design/pnc.md

```diff
@@ -0,0 +1,192 @@
+   - `publicUrl` key is used to get download url
+   - supported checksums are saved
+   - build ids are saved
+3) Download queue is formed containing above info for each artifact
+4) SBOM component info for each artifact is fetched from PNC @ `/artifacts/{id}`, `purl` key
+5) Queue is executed and each artifact is downloaded, checksum verified
+6) SBOM is compiled, including PNC components from step 4)
+
+
+### Missing features
+
+PNC SBOM components don't have the most complete purl, e.g.:
+`pkg:maven/org.kie.trustyai/explainability-service@999.0.0.managedsvc-redhat-00001?type=zip`
+
+According to [purl spec](https://github.com/package-url/purl-spec/blob/master/PURL-TYPES.rst#maven), `classfier` and
+`type` parameters should be also defined, method is currently pending depending on if the artifacts are single components,
+or products that contain other components. Purl should also specify a `repository_url` parameter.
+
+
+
+## Design for the implementation in Cachi2
+
+### PNC config
+Existing `fetch-artifacts-pnc.yaml` file ([docs](https://osbs.readthedocs.io/en/osbs_ocp3/users.html?highlight=fetch#fetch-artifacts-pnc-yaml))
+will be used as configuration. It needs to be extended to also include PNC instance url.
+
+### Pre-fetch PNC dependencies
+Cachi2 will use the `artifacts` key specified in the `fetch-artifacts-pnc.yaml` file to fetch info about given artifacts
+from PNC (`/artifacts/{id}` endpoint), including url to download the artifact, checksums to verify integrity, and the purl.
+The artifacts should be fetched into a separate subdirectory in cachi2 output folder, and into the target path.
+
+#### Fetching sources
+Cachi2 will use the `build_id` keys to fetch build sources from PNC (`/builds/{id}/scm-archive` endpoint). These sources
+should be fetched into a separate subdirectory in cachi2 output folder, different from the artifacts.
+
+### Generating the SBOM
+As mentioned in [missing features](#missing-features), the purl needs to specify more parameters
+- `repository_url` - obtainable by using host part of `publicUrl` and the path in `targetRepository.repositoryPath`
+  from PNC `/artifacts/{id}` endpoint response.
+- `type` & `classifier` - solution pending
+
+Here's an example purl that should be generated:
+```
+pkg:maven/org.example/package@1.2.3.org-1?type=zip&classifier=dist&repository_url=https://repo.maven.apache.org/maven2
+```
+
+#### Relevant configuration for the build
+No environment variables need to be set.
+
+#### Expected location of artifacts
+The current implementation expects artifacts in the `artifacts/` path in the root of repository.
+Due to cachi2's usual design patterns, the artifacts will remain in cachi2's default output directory
+(`./cachi2-output`), and users will be required to act accordingly.
```

Comment:
Could we at least create a symlink `artifacts -> .cachi2-output`?

Reply:
Since this is a Konflux-specific issue, I believe we could treat it on the pipeline level, so possibly a suggestion for [build-definitions](https://github.com/konflux-ci/build-definitions).

Reply:
Yeah, I would prefer this not be handled in cachi2 if possible. As a general principle, I think we should strive to avoid modifying the user repository unless absolutely necessary for our core functionality